packages/catlog/src/tt/context.rs

```diff
@@ -0,0 +1,50 @@
+//! Contexts store the type and values of in-scope variables during elaboration
+use crate::tt::{prelude::*, val::*};
+
+/// The variable context during elaboration
+pub struct Context {
+    /// Stores the value of each of the variables in context
+    pub env: Env,
+    /// Stores the names and types of each of the variables in context.
+    ///
+    /// We allow the type to be "none" as a hack for the `self` variable before we
+    /// know the type of the `self` variable.
+    pub scope: Vec<(VarName, LabelSegment, Option<TyV>)>,
```

Comment:
IMO, would increase legibility to replace this triple with a struct, something like the following if I'm understanding the meaning each of field correctly:

```rust
struct VarInContext {
  name: VarName,
  label: LabelSegment,
  type: Option<TyV>
}
```

Also feels slightly incongruous to have `VarName` as a type alias for `NameSegment` but use `LabelSegment` directly. Why not just use `NameSegment` directly?

Reply:
The answer is that I wanted to be self-documenting when I was originally writing this, but then I got lazy when I was adding labels everywhere...

I'm happy to have `VarInContext` though.

Reply:
If you find this inconsistency ugly, I can add `VarLabel` and `FieldLabel` aliases.